DESIGN_GUIDELINES.md

```diff
@@ -0,0 +1,1050 @@
+└── tests/                  # Test directory
+```
+
+## Code Organization
+
+1. **Separation of Concerns**:
+   - `models.py`: Define data models and validation logic
+   - `server.py`: Implement MCP server, tools, and resources
+   - `consts.py`: Define constants used across the server
+   - Additional modules for specific functionality (e.g., API clients)
+
+2. **Keep modules focused and limited to a single responsibility**
+
+3. **Use clear and consistent naming conventions**
+
+### Entry Points
+
+MCP servers should follow these guidelines for application entry points:
+
+1. **Single Entry Point**: Define the main entry point only in `server.py`
+   - Do not create a separate `main.py` file
+   - This maintains clarity about how the application starts
+
+2. **Main Function**: Implement a `main()` function in `server.py` that:
+   - Handles command-line arguments
+   - Sets up environment and logging
+   - Initializes the MCP server
+   - Starts the server with appropriate transport
+
+Example:
+
+```python
+def main():
+    """Run the MCP server with CLI argument support."""
+    parser = argparse.ArgumentParser(description='A Model Context Protocol (MCP) server')
+    parser.add_argument('--sse', action='store_true', help='Use SSE transport')
+    parser.add_argument('--port', type=int, default=8888, help='Port to run the server on')
+
+    args = parser.parse_args()
+
+    # Run server with appropriate transport
+    if args.sse:
+        mcp.settings.port = args.port
+        mcp.run(transport='sse')
+    else:
+        mcp.run()
+
+
+if __name__ == '__main__':
+    main()
+```
+
+3. **Package Entry Point**: Configure the entry point in `pyproject.toml`:
+
+```toml
+[project.scripts]
+"awslabs.your-mcp-server" = "awslabs.your_mcp_server.server:main"
+```
+
+## Package Naming and Versioning
+
+1. **Package Naming**: Follow the established naming pattern:
+   - Namespace: `awslabs`
+   - Package name: lowercase with hyphens (in pyproject.toml)
+   - Python module: lowercase with underscores
+
+   Example:
+   ```toml
+   # In pyproject.toml
+   name = "awslabs.nova-canvas-mcp-server"
+   ```
+
+   ```python
+   # In Python imports
+   from awslabs.nova_canvas_mcp_server import models
+   ```
+
+2. **Versioning**: Store version information in `__init__.py`:
+
+   ```python
+   # awslabs/your_mcp_server/__init__.py
+   """awslabs Your MCP Server."""
+
+   __version__ = "0.1.0"
+   ```
+
+3. **Version Synchronization**: Our monorepo `release.py` bumps the patch version upon changes in:
+   - `pyproject.toml`
+   - `__init__.py` in the package
+   - Configure `commitizen` in `pyproject.toml` to update versions automatically
+
+   ```toml
+   [tool.commitizen]
+   name = "cz_conventional_commits"
+   version = "0.0.0"
+   tag_format = "$version"
+   version_files = [
+       "pyproject.toml:version",
+       "awslabs/your_mcp_server/__init__.py:__version__"
+   ]
+   update_changelog_on_bump = true
+   ```
+
```

Comment:
```suggestion
   ```

   _NOTE: commitizen attempts to use remote tags, and since each package may be a different version, this feature is disabled in our monorepo_
  
```